### ISSUE TITLE
Changes in training code to send both reference images and instruction to MLLM

### ISSUE THREAD

**Issue by Vinayak2104** [VectorSpaceLab/OmniGen2#109]
input_images = batch['input_images']
            output_image = batch['output_image']
            text_mask = batch['text_mask']
            text_input_ids = batch['text_ids']
            input_images_pil = batch['input_images_pil']
            def wrap_with_image_tags(instruction: str, num_images: int):
                # builds exactly: "<img1>:<|vision_start|>…<imgN>:…"
                tags = "".join(
                    f"<img{i}>:<|vision_start|><|image_pad|><|vision_end|>"
                    for i in range(1, num_images+1)
                )
                # then wrap into the chat template:
                return (
                    "<|im_start|>system\n"
                    "You are a helpful assistant that generates high-quality images based on user instructions."
                    "<|im_end|>\n"
                    "<|im_start|>user\n"
                    f"{tags}{instruction}"
                    "<|im_end|>\n"
                    "<|im_start|>assistant\n")
            
            with accelerator.accumulate(model):
                with torch.no_grad():
                    raw_texts = [text_tokenizer.decode(ids, skip_special_tokens=False) for ids in batch["text_ids"]]
                    prompts = [wrap_with_image_tags(text, len(input_images_pil[i]))for i, text in enumerate(raw_texts)]
                    max_pos = getattr(mllm.config, "max_position_embeddings", 2048)
                    tokenized = processor.tokenizer(prompts,padding=True,truncation=True,max_length=max_pos,return_tensors="pt")
                    prompts = [processor.tokenizer.decode(ids, skip_special_tokens=False)for ids in tokenized.input_ids]
                    inputs = processor(text=prompts, images=input_images_pil, return_tensors="pt", padding=True)
                    embed_device = mllm.model.embed_tokens.weight.device
                    inputs = {k: v.to(embed_device) for k, v in inputs.items()}
                    text_feats = mllm(**inputs,output_hidden_states=True)
                    text_feats = text_feats.hidden_states[-1]
                    text_mask  = inputs["attention_mask"]
                    text_feats = text_feats.to(accelerator.device)
                    text_mask  = text_mask.to(accelerator.device)

i have done some changes in the way text_feats are extracted but upon running this code i am getting an out of index error in RoPE,any suggestions on how to fix this?
